dataactcore/utils/jobQueue.py

```diff
@@ -22,6 +30,32 @@ def enqueue(jobID):
     return response.json()
 
 
+def generate_f_file(submission_id, job_id, interface_holder_class,
+                    timestamped_name, is_local):
+    """Write rows from fileF.generateFRows to an appropriate CSV. Here the
+    third parameter, interface_holder_class, is a bit of a hack. Importing
+    InterfaceHolder directly causes cyclic dependency woes, so we're passing
+    in a class"""
+    with job_app.app_context():
+        job_manager = interface_holder_class().jobDb
+
+        try:
+            rows_of_dicts = fileF.generateFRows(job_manager.session,
+                                                submission_id)
+            header = [key for key in fileF.mappings]    # keep order
+            body = []
+            for row in rows_of_dicts:
+                body.append([row[key] for key in header])
+
+            write_csv(timestamped_name, is_local, header, body)
+            job_manager.markJobStatus(job_id, "finished")
```

Comment:
This is a good opportunity to think about the impending refactor of db access for code running as part of a Flask request.

The first pass we discussed was to take all the methods in the interface classes (_i.e._, `job_manager` in this case) and move them to one big file in `dataactcore.interfaces`. That will be an ugly-looking file, but will provide a starting point to putting things in better places (models, etc. @nmonga91 had some good ideas about this).

So in this case, you'd end up calling `markJobStatus(job_id, "finished")` without the need to get a `job_manager` first. 

Alternately, for these short functions, we should ask ourselves--do we really need to abstract to a function that is (or could be) 1-2 lines of code? Are there ways to tweak the models themselves that would support this effort?

If we opt for the latter, you could get ahead of the game by getting rid of the `markJobStatus` call and doing that inline.

Paging @nmonga91 and @jworcestBAH on this one!